### PR TITLE
[Merged by Bors] - Sync tortoise integration 

### PR DESCRIPTION
**Pull request by y0sher**
## Motivation
this merges the new sync code with the new verifying tortoise code. 

TODO: generate new sync database,

### CODE REVIEW

**Comment by y0sher** [comment]
bors try

**Comment by y0sher** [comment]
bors try

**Comment by y0sher** [comment]
bors try

**Comment by y0sher** [comment]
bors try

**Comment by y0sher** [comment]
bors try

**Comment by y0sher** [comment]
bors try

**Comment by y0sher** [comment]
bors try

**Comment by y0sher** [comment]
bors try

**Comment by antonlerner** [comment]
bors try

**Comment by antonlerner** [comment]
bors try

**Comment by antonlerner** [comment]
bors try

**Comment by antonlerner** [comment]
bors try

**Comment by antonlerner** [comment]
bors try

**Comment by antonlerner** [comment]
bors try

**Comment by noamnelke** [comment]
bors try-

**Comment by noamnelke** [comment]
bors try-

**Comment by y0sher** [comment]
bors merge

File: mesh/meshdb.go [hunk 1]
```diff
@@ -97,6 +97,9 @@ func NewPersistentMeshDB(path string, blockCacheSize int, log log.Log) (*DB, err
 			log.With().Error("Error inserting genesis block to db", blk.ID(), blk.LayerIndex)
 		}
 	}
+	if err := ll.SaveLayerInputVectorByID(GenesisLayer().Index(), types.BlockIDs(GenesisLayer().Blocks())); err != nil {
+		log.With().Error("Error inserting genesis input vector to db", GenesisLayer().Index())
+	}
 	return ll, nil
```
Comment (antonlerner) [inline mesh/meshdb.go:103]: `return ll, err`

**Review by antonlerner — APPROVED** [review]

**Comment by antonlerner** [comment]
bors merge

**Pull request closed without merging**